File: bindings/dotnet/src/Zlink/Runtime/Sockets/SubSocket.cs
// SPDX-License-Identifier: MPL-2.0


namespace Systems.Zlink;

internal sealed class SubSocket : SubscriberSocketBase, ISubSocket
{
    public SubSocket(Context context)
        : base(context, SocketType.Sub)
    {
        Options = new SubSocketOptions(this);
    }

    public new SubSocketOptions Options { get; }

    public void SetRoutingId(RoutingId routingId)
    {
        Kernel.SetOption(SocketOptions.RoutingId, routingId.ToBytes());
    }

    public RoutingId GetRoutingId()
    {
        return RoutingId.From(Kernel.GetOption(SocketOptions.RoutingId));
    }

}

File: bindings/node/src/zlink/runtime/sockets/sub_socket.ts
// SPDX-License-Identifier: MPL-2.0

import { SubSocketOptions } from './socket_options';
import {
  SubscriberSocket,
} from './socket_operations';
import type { RuntimeContext as Context } from '../core/context';
import { SocketType as NativeSocketType } from '../../contracts/sockets/socket_constants';

export class SubSocket extends SubscriberSocket {
  readonly options: SubSocketOptions;
  constructor(ctx: Context) { super(ctx, NativeSocketType.SUB); this.options = SubSocketOptions.create(this); }
}
